docs/guide/developing/resources/create.md

```diff
@@ -0,0 +1,110 @@
+# Create
+
+## Description
+
+Device with collection resource allows dynamic creation of resources of selected type.
+
+## Guide
+
+To develop your own device you can examine the example in [cloud_server](https://github.com/iotivity/iotivity-lite/blob/master/apps/cloud_server.c). Lets examine the example to identify the necessary steps that allow a device to dynamically create resources.
+
+1. Create a collection resource
+```C
+  oc_resource_bind_resource_type(col, "oic.wk.col");
+```
+2. Determine which resource types can populate the collection
+```C
+  oc_collection_add_supported_rt(col, "oic.r.switch.binary");
+  oc_collection_add_mandatory_rt(col, "oic.r.switch.binary");
+```
+3. Enable create operation on the collection resource
+```C
+oc_resource_bind_resource_interface(col, OC_IF_CREATE);
+```
+4. Define constructor and destructor functions for collection item instances
+```C
+oc_collections_add_rt_factory("oic.r.switch.binary", get_switch_instance, free_switch_instance);
```

Comment:
@Danielius1922 can you include code of `get_switch_instance` and rename it to `new_switch_instance`. Same for `free_switch_instance`, include code here. Thanks

Reply:
it's pretty long though, maybe I can trim it down only to the most important parts